src/server/game/Entities/Player/Player.cpp

```diff
@@ -23544,9 +23544,16 @@ void Player::UpdateObjectVisibility(bool forced, bool fromUpdate)
 
 void Player::UpdateVisibilityForPlayer(bool mapChange)
 {
+    // After added to map seer must be a player - there is no possibility to still have different seer (all charm auras must be already removed)
+    if (mapChange && m_seer != this)
+    {
+        m_seer = this;
```

Comment:
TODO: add an error log here (I've tried doing it via GitHub but the file `Player.cpp` is too big and it does not work)